src/fondant/pipeline.py

```diff
@@ -303,240 +303,3 @@ def _validate_pipeline_definition(self, run_id: str):
-                component_args = fondant_component_op.arguments
-
-                if previous_component_task is not None:
-                    component_task = kubeflow_component_op(
-                        input_manifest_path=manifest_path,
-                        metadata=metadata,
-                        **component_args,
-                    )
-                else:
-                    metadata = json.dumps(
-                        {"base_path": self.base_path, "run_id": run_id},
-                    )
-                    # Add metadata to the first component
-                    component_task = kubeflow_component_op(
-                        input_manifest_path=manifest_path,
-                        metadata=metadata,
-                        **component_args,
-                    )
-                    metadata = ""
-                # Set optional configurations
-                component_task = _set_task_configuration(
-                    component_task,
-                    fondant_component_op,
-                )
-                # Set the execution order of the component task to be after the previous
-                # component task.
-                if previous_component_task is not None:
-                    component_task.after(previous_component_task)
-
-                # Update the manifest path to be the output path of the current component task.
-                manifest_path = component_task.outputs["output_manifest_path"]
-
-                previous_component_task = component_task
-
-            return pipeline
-
-        logger.info(f"Compiling pipeline: {self.name}")
-
-        kfp.compiler.Compiler().compile(pipeline, self.package_path)
-
-        logger.info("Pipeline compiled successfully")
-
-    def __repr__(self) -> str:
-        """Return a string representation of the FondantPipeline object."""
-        return f"{self.__class__.__name__}({self._graph!r}"
-
-
-class Client:
-    """Class representing a Fondant Client."""
-
-    def __init__(self, host: str):
-        """
-        Args:
-            host: The `host` URL argument specifies the Kubeflow Pipelines API endpoint to
-             which the client should send requests.
-        """
-        self.host = host
-        self.client = kfp.Client(host=self.host)
-
-    def get_pipeline_id(self, pipeline_name: str) -> str:
-        """
-        Function that returns the id of a pipeline given a pipeline name
-        Args:
-            pipeline_name: the name of the pipeline
-        Returns:
-            The pipeline id.
-        """
-        return self.client.get_pipeline_id(pipeline_name)
-
-    def get_pipeline_version_ids(self, pipeline_id: str) -> t.List[str]:
-        """Function that returns the versions of a pipeline given a pipeline id."""
-        pipeline_versions = self.client.list_pipeline_versions(pipeline_id).versions
-        return [version.id for version in pipeline_versions]
-
-    def delete_pipeline(self, pipeline_name: str):
```

Comment:
Where do you see all the other functionalities that are not related to compiling and running? user might still want to list pipeline runs, delete pipelines, ...

Reply:
Not sure if we need to wrap this functionality for every platform we want to run on. At least at this point in time, I would be ok with removing it and having the user work with the underlying platform directly.

Reply:
makes sense, we'll re-add it later if we notice that it's needed